src/oauth-tooling.ts

```diff
@@ -123,29 +138,39 @@ function buildRequestAccessTokenUrl(accessTokenEndpoint: string, queryParams?: O
  *
  * @param tokenInfoUrl
  * @param accessToken
+ * @param logger - optional logger
  * @returns {Promise<Token>}
  */
-function getTokenInfo(tokenInfoUrl: string, accessToken: string): Promise<Token> {
+function getTokenInfo(tokenInfoUrl: string, accessToken: string, logger?: Logger): Promise<Token> {
 
-    const promise = fetch(`${tokenInfoUrl}?access_token=${accessToken}`)
-    .then((response) => {
+  const logOrNothing = safeLogger(logger);
 
-      const status = response.status;
+  const promise = fetch(`${tokenInfoUrl}?access_token=${accessToken}`)
+  .then((response) => {
 
-      return response.json()
-      .then((data) => {
+    const status = response.status;
 
-        if (response.status === HttpStatus.OK) {
-          return data;
-        } else {
-          return Promise.reject({ status, data });
-        }
-      });
-    })
-    .catch((err) => Promise.reject({
-      msg: `Error requesting tokeninfo from ${tokenInfoUrl}`,
-      err
-    }));
+    return response.json()
+    .then((data) => {
+
+      if (status === HttpStatus.OK) {
```

Comment:
Is this a candidate for its own function. Since there was a discussion to have logging at more places (every one that does http stuff).

Reply:
You mean this whole block? Yeah in general yes. However we currently have 2 places I think and you need context information, you would have to pass that to the function. So yes would be possible, not sure whether worth it.

Reply:
is that context information available in the Promise value, here called `data`

Reply:
we also want to log the url we request for example